sagemaker-python-sdk/mxnet_gluon_mnist/mxnet_mnist_with_gluon_1.3.ipynb

```diff
@@ -0,0 +1,1139 @@
+   "metadata": {},
+   "source": [
+    "### Overview\n",
+    "\n",
+    "MNIST is a widely used dataset for handwritten digit classification. It consists of 70,000 labeled 28x28 pixel grayscale images of hand-written digits. The dataset is split into 60,000 training images and 10,000 test images. There are 10 classes (one for each of the 10 digits). This tutorial will show how to train and test an MNIST model on SageMaker local mode using MXNet and the Gluon API."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Instance type = local_gpu\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "import subprocess\n",
+    "import boto3\n",
+    "import sagemaker\n",
+    "from sagemaker.mxnet import MXNet\n",
+    "from mxnet import gluon\n",
+    "from sagemaker import get_execution_role\n",
+    "\n",
+    "sagemaker_session = sagemaker.Session()\n",
+    "\n",
+    "instance_type = 'local'\n",
+    "\n",
+    "if subprocess.call('nvidia-smi') == 0:\n",
+    "    ## Set type to GPU if one is present\n",
+    "    instance_type = 'local_gpu'\n",
+    "    \n",
+    "print(\"Instance type = \" + instance_type)\n",
+    "\n",
+    "role = get_execution_role()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Download training and test data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<mxnet.gluon.data.vision.datasets.MNIST at 0x7fdf784e3290>"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "gluon.data.vision.MNIST('./data/train', train=True)\n",
+    "gluon.data.vision.MNIST('./data/test', train=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Uploading the data\n",
+    "\n",
+    "We use the `sagemaker.Session.upload_data` function to upload our datasets to an S3 location. The return value `inputs` identifies the location -- we will use this later when we start the training job."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inputs = sagemaker_session.upload_data(path='data', key_prefix='data/mnist')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Implement the training function\n",
+    "\n",
+    "We need to provide a training script that can run on the SageMaker platform. The training scripts are essentially the same as one you would write for local training, except that you need to provide a `train` function. When SageMaker calls your function, it will pass in arguments that describe the training environment. Check the script below to see how this works.\n",
```

Comment:
this paragraph does not correspond to the new training script format (script mode)